conans/test/integration/command_v2/list_package_ids_test.py

```diff
@@ -0,0 +1,277 @@
+
+
+class TestListPackagesFromRemotes(TestListPackageIdsBase):
+    def test_by_default_search_only_in_cache(self):
+        self._add_remote("remote1")
+        self._add_remote("remote2")
+
+        expected_output = textwrap.dedent("""\
+        Local Cache:
+          There are no matching references
+        """)
+
+        self.client.run(f"list package-ids {self._get_fake_recipe_refence('whatever/0.1')}")
+        assert expected_output == self.client.out
+
+    def test_search_no_matching_recipes(self):
+        self._add_remote("remote1")
+        self._add_remote("remote2")
+
+        expected_output = textwrap.dedent("""\
+        Local Cache:
+          There are no matching references
+        remote1:
+          There are no matching references
+        remote2:
+          There are no matching references
+        """)
+
+        rrev = self._get_fake_recipe_refence('whatever/0.1')
+        self.client.run(f"list package-ids -c -a {rrev}")
+        assert expected_output == self.client.out
+
+    def test_fail_if_no_configured_remotes(self):
+        self.client.run("list package-ids -a whatever/1.0", assert_error=True)
+        assert "ERROR: The remotes registry is empty" in self.client.out
+
+    def test_search_disabled_remote(self):
+        self._add_remote("remote1")
+        self.client.run("remote disable remote1")
+        rrev = self._get_fake_recipe_refence('whatever/0.1')
+        self.client.run(f"list package-ids -r remote1 {rrev}", assert_error=True)
+        assert "ERROR: Remote 'remote1' is disabled" in self.client.out
+
+
+class TestRemotes(TestListPackageIdsBase):
+
+    def test_search_with_full_reference_but_no_packages_in_cache(self):
+        self.client.save({
+            "conanfile.py": GenConanfile("test_recipe", "1.0.0").with_package_file("file.h", "0.1")
+        })
+        self.client.run("export . user/channel")
+        rrev = self._get_lastest_recipe_ref("test_recipe/1.0.0@user/channel")
+        self.client.run(f"list package-ids {repr(rrev)}")
+        expected_output = textwrap.dedent(f"""\
+        Local Cache:
+          There are no matching references
+        """)
+        assert expected_output == str(self.client.out)
+
+    def test_search_with_full_reference(self):
+        remote_name = "remote1"
+        recipe_name = "test_recipe/1.0.0@user/channel"
+        self._add_remote(remote_name)
+        self._upload_full_recipe(remote_name, recipe_name)
+        rrev = self._get_lastest_recipe_ref(recipe_name)
+        self.client.run(f"list package-ids -r remote1 {repr(rrev)}")
+
+        expected_output = textwrap.dedent("""\
+        remote1:
+          %s:.{40}
+            settings:
+              arch=.*
+              build_type=.*
+              os=.*
+            options:
+              shared=False
+            requires:
+              pkg/0.1@user/channel:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
+        """ % repr(rrev))
```

Comment:
Why `%` formatting?

Reply:
If you realize I have one line like this `%s:.{40}` so I don't want to have this `40` as a `f-string` value but a regex one